src/app/pages/user/dialogs/base/user-details-form/user-details-form.html

```diff
@@ -0,0 +1,82 @@
+            autocomplete="username" name="username" required>
+          <button type="button" tabindex=-1 mat-button *ngIf="data.username" matSuffix mat-icon-button aria-label="Clear"
+            (click)="data.username=''">
+            <mat-icon>close</mat-icon>
+          </button>
+          <mat-error *ngIf="userNameFormControl.hasError('required')">שם משתמש הינו שדה חובה</mat-error>
+          <mat-error *ngIf="userNameFormControl.hasError('minlength') || userNameFormControl.hasError('pattern')">שם
+            משתמש חייב להיות לפחות חמישה תווים באנגלית</mat-error>
+        </mat-form-field>
+      </div>
+      <div class="msw-form-row">
+        <mat-form-field color="primary">
+          <mat-select placeholder="סוג משתמש" formControlName="role" [(ngModel)]="data.role" name="role"
+            (valueChange)="onUserTypeChange($event)" autocomplete='role' required>
+            <mat-option disabled>בחר</mat-option>
+            <mat-option *ngFor="let _role of roles" [value]="_role">
+              {{userRoleEnum[_role]}}
+            </mat-option>
+          </mat-select>
+          <mat-error *ngIf="selectUserType.hasError('required')">אנא בחר סוג משתמש</mat-error>
+        </mat-form-field>
+      </div>
+    </div>
+    <div class="msw-div-column-left">
+      <div class="msw-form-row">
+        <mat-form-field color="primary">
+          <input matInput #input formControlName="lastname" placeholder="שם משפחה" [(ngModel)]="data.lastname"
+            autocomplete='family-name' name="lastname" required>
+          <button type="button" tabindex=-1 mat-button *ngIf="data.lastname" matSuffix mat-icon-button aria-label="Clear"
+            (click)="data.lastname=''">
+            <mat-icon>close</mat-icon>
+          </button>
+          <mat-error *ngIf="formControl.invalid">שם משפחה הינו שדה חובה</mat-error>
+        </mat-form-field>
+      </div>
+      <div class="msw-form-row">
+        <mat-form-field color="primary">
+          <input matInput #input formControlName="email" placeholder="אימייל" [(ngModel)]="data.email" name="email"
+            autocomplete='email' required email>
+          <button type="button" tabindex=-1 mat-button *ngIf="data.email" matSuffix mat-icon-button aria-label="Clear"
+            (click)="data.email=''">
+            <mat-icon>close</mat-icon>
+          </button>
+          <mat-error *ngIf="EmailFormControl.hasError('required')">אימייל הינו שדה חובה</mat-error>
+          <mat-error *ngIf="EmailFormControl.hasError('email')">אימייל לא חוקי</mat-error>
+        </mat-form-field>
+      </div>
+      <div class="msw-form-row">
+        <mat-form-field color="primary">
+          <mat-select placeholder="כיתה" formControlName="class" [(ngModel)]="data.class" name="class" [compareWith]="compareClass">
+           <mat-option disabled>בחר</mat-option>
+           <mat-option *ngFor="let class1 of classes" [value]="class1">
+              {{class1.name}}
```

Comment:
hey, `class1`?

Reply:
I agree about distinguishing. But not with class1. Let's find better naming for that